src/Data.jl

```diff
@@ -0,0 +1,97 @@
+module Data
+
+export ProgramSynthesisProblem, InputOutputProblem
+
+abstract type ProgramSynthesisProblem end
+
+"""
+A program synthesis problem that is represented in the form of input-output examples.
+"""
+struct InputOutputProblem <: ProgramSynthesisProblem
```

Comment:
Let's go for data structures that reflect individual examples:
 - We need an abstract type `Example`
 - Then make struct `IOExample` of type Example
 - then struct `Problem` (or maybe better named `Task`) can contain a vector of `Example` instances

Reply:
We can also probably leave out 'ProgramSynthesis' from the names of structs. The fact that they are in a program synthesis library indicates their intended purpose :)